packages/flutter_tools/lib/src/android/gradle_utils.dart

```diff
@@ -204,20 +204,28 @@ distributionUrl=https\\://services.gradle.org/distributions/gradle-$gradleVersio
 /// The Android plugin version is specified in the [build.gradle] file within
 /// the project's Android directory.
 String getGradleVersionForAndroidPlugin(Directory directory, Logger logger) {
-  final File buildFile = directory.childFile('build.gradle');
+  const String buildFileName = 'build.gradle/build.gradle.kts';
+
+  final File buildFile;
+  if (directory.childFile('build.gradle.kts').existsSync()) {
```

Comment:
This logic did not get updated with the new prefer groovy behavior. 

Blocking

Reply:
Good catch, thank you. I'll add tests to make sure it won't happen in the future.